cosmic-core/server/src/main/java/com/cloud/network/NetworkServiceImpl.java

```diff
@@ -3337,7 +3337,7 @@ public boolean releaseSecondaryIpFromNic(final long ipAddressId) {
                 }
             }
             //check if the secondary ip associated with any static nat rule
-            final IPAddressVO publicIpVO = _ipAddressDao.findByVmIp(secondaryIp);
+            final IPAddressVO publicIpVO = _ipAddressDao.findByIpAndNetworkId(secIpVO.getNetworkId(), secondaryIp);
```

Comment:
On line 3326 there's a check for `Isolated` networks. Should that be fixed the VPC tiers as well?